src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts

```diff
@@ -127,16 +124,58 @@ export class HierarchicalDataService implements SamHiercarchicalServiceInterface
   }
 
   getHiercarchicalById(item: SamHiercarchicalServiceSearchItem): Observable<SamHiercarchicalServiceResult> {
```

Comment:
@thomastighe Can you split this into a couple function calls for clarity and maintainability down the road? It almost looks like you could have a separate, small class for some of the logic in here.

Reply:
Cleaned up and split to a new file and reused